### PR TITLE
Add find_project_root function

### PR DESCRIPTION
**Pull request by romasku**
This is part of https://github.com/neuromation/neuro-extras/issues/31

### CODE REVIEW

File: neuromation/api/__init__.py [hunk 1]
```diff
@@ -145,7 +146,7 @@ async def login(
     *,
     url: URL = DEFAULT_API_URL,
     path: Optional[Path] = None,
-    timeout: aiohttp.ClientTimeout = DEFAULT_TIMEOUT
+    timeout: aiohttp.ClientTimeout = DEFAULT_TIMEOUT,
```
Comment (romasku) [inline neuromation/api/__init__.py:149]: I think this changes were introduced by `make fmt`. Should I undo them?

Comment (serhiy-storchaka) [reply neuromation/api/__init__.py:148]: It looks like a glitch of **black**. If repeated `make fmt` does not change the code you can keep it.

File: neuromation/api/__init__.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import pathlib
```
[the remark this answers is not in the document]
Comment (romasku) [reply neuromation/api/__init__.py]: Fixed.

File: neuromation/api/__init__.py [hunk 3]
```diff
@@ -165,10 +165,24 @@ async def login_headless(
...
 
 async def logout(*, path: Optional[Path] = None) -> None:
     await Factory(path).logout()
+
+
+class NoProjectRoot(Exception):
+    pass
+
+
+def find_project_root() -> Path:
```
Comment (asvetlov) [inline neuromation/api/__init__.py:181]: Please reuse this function in `load_user_config()` to reduce the code duplication.
Another reason is the future change for search criteria. Perhaps we want to support `.neuro` folder with toml file inside as an option to keep the single space for configs. `neuro-flow` keeps its own scenarios in this way. 
The `.neuro` folder support can wait for a separate PR.

File: neuromation/api/__init__.py [hunk 4]
```diff
@@ -165,10 +165,24 @@ async def login_headless(
...
+    here = Path.cwd()
```
Comment (asvetlov) [inline neuromation/api/__init__.py:182]: Please consider optional passing a directory as a starting point. The function signature can look like `def find_project_root(cwd: Optional[Path] = None) -> Path: ...`

File: neuromation/api/__init__.py [hunk 5]
```diff
@@ -165,10 +165,24 @@ async def login_headless(
...
+    while here.parent != here:
+        config = here / ".neuro.toml"
+        if config.exists():
+            return here
+        here = here.parent
+    raise NoProjectRoot(f"Project root is not found for {here}")
```
Comment (asvetlov) [inline neuromation/api/__init__.py:188]:
```suggestion
    raise ConfigError(f"Project root is not found for {here}")
```

**Comment by asvetlov** [comment]
Ooops, you've merged the PR just when I was writing the review :)
Please fix notes in the follow-up PR (or convince me that these notes are not relevant).

**Comment by romasku** [comment]
Oh, sorry, I'll do follow-up PR.

**Comment by asvetlov** [comment]
No problem

**Pull request merged**